ddb-popper/ddb-popper.js

```diff
@@ -0,0 +1,200 @@
+        let ddbURL;
+
+        try {
+            ddbURL = await game.settings.get(DDBPopper.CONFIG.moduleName, data.actor._id);
+        } catch (e) {
+            if(e.message == "This is not a registered game setting") {
+                await game.settings.register(DDBPopper.CONFIG.moduleName, data.actor._id, DDBPopper.SETTINGS_META);
+                ddbURL = await game.settings.get(DDBPopper.CONFIG.moduleName, data.actor._id);
+            } else {
+                ddbURL = null;
+                throw(e);
+            }
+        } finally {
+            return ddbURL;
+        }
+    }
+
+    /**
+     * Adds the DDB button to the supplied html
+     * @param {Object} html -- the html of the ActorSheet5eCharacter
+     * @param {Object} data -- the data of the actor
+     */
+    async _addDDBButton (html, data) {
+        /**
+         * Finds the header and the close button
+         */
+        const windowHeader = html.parent().parent().find(".window-header");
+        const windowCloseBtn = windowHeader.find(".close");
+
+        /**
+         * jquery reference to the D&D Beyond button to add to the sheet
+         */
+        const ddbButton = $(
+            `<a class="ddb-popup">
+                <img src=${DDBPopper.CONFIG.ddbLogo} style=${DDBPopper.CONFIG.imgStyle}> DDB
+            </a>`
+        );
+        
+        /**
+         * Remove any existing instances of the button on the sheet
+         */
+        windowHeader.find('.ddb-popup').remove();
+        windowCloseBtn.before(ddbButton);
+
+        /**
+         * When the DDB button is clicked, lookup the DDB URL, then look for an existing popup,
+         * and if none exists, and the DDB URL is null, open the form to set the URL, 
+         * otherwise if the popup does not exist but the DDB URL is not null, create the popup
+         * or else simply focus the existing popup
+         */
+        ddbButton.click(async ev => {
+            ev.preventDefault();
+            const actorDDBURL = await this._getActorDDBURL(data);
+
+            if (actorDDBURL == null || actorDDBURL.length == 0){
+                new DDBURLEntryForm(actorDDBURL, data, {closeOnSubmit: true}).render(true);
+            } else if ((this.existingPopup == null || this.existingPopup.closed) && actorDDBURL.length > 0) {
+                this.existingPopup = window.open(actorDDBURL, DDBPopper.CONFIG.moduleName, DDBPopper.CONFIG.windowFeatures);
+            } else if (this.existingPopup && actorDDBURL.length > 0) {
+                this.existingPopup.focus();
+            }
+        });
+
+        /**
+         * When the DDB button is right-clicked, lookup the DDB URL for the actor,
+         * then open the form to set the URL 
+         */
+        ddbButton.contextmenu(async ev => {
+            ev.preventDefault();
+            const actorDDBURL = await this._getActorDDBURL(data);
+
+            new DDBURLEntryForm(actorDDBURL, data, {closeOnSubmit: true}).render(true);
```

Comment:
Duplicated above, create helper function

Reply:
I waffled on this one, because all I'm doing here is instantiating the other module class. Do you really think a helper is the best idea in that scenario?